.storybook/preview.tsx

```diff
@@ -0,0 +1,78 @@
+import type { Preview } from '@storybook/react';
+import '../app/globals.scss';
+import React, { Suspense, useState, useEffect } from 'react';
+import { AbstractIntlMessages, NextIntlClientProvider } from 'next-intl';
+import { type Locales, getDirection } from '../lib/localisation/locales';
+
+const loadMessages = async (
+  locale: Locales,
+): Promise<AbstractIntlMessages | undefined> => {
+  try {
+    const messages = (
+      await import(`../lib/localisation/messages/${locale}.json`)
+    ).default;
+    return messages;
+  } catch (error) {
+    console.error(`Could not load messages for locale ${locale}`, error);
+    return undefined;
+  }
+};
+
+const withIntl = (Story, context) => {
+  const [messages, setMessages] = useState<AbstractIntlMessages | undefined>(
+    undefined,
+  );
```

Comment:
It's probably nothing but I usually write this as `const [messages, setMessages] = useState<AbstractIntlMessages>();`. Please ignore this if you think this is unnecessary.